dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEventDataQueryService.java

```diff
@@ -223,9 +229,78 @@ public EventQueryParams getFromRequest(EventDataQueryRequest request, boolean an
       eventQueryParams = builder.build();
     }
 
+    validateQueryParamsForConfidentialAndSkipAnalytics(eventQueryParams);
+
     return eventQueryParams;
   }
 
+  private void validateQueryParamsForConfidentialAndSkipAnalytics(
+      EventQueryParams eventQueryParams) {
+    if (eventQueryParams.includeConfidentialOrSkipAnalyticsItems()) {
+      return;
+    }
+    Set<TrackedEntityAttribute> confidentialAttributes =
+        Stream.concat(
+                eventQueryParams.getItems().stream(), eventQueryParams.getItemFilters().stream())
+            .map(QueryItem::getItem)
+            .filter(Objects::nonNull)
+            .filter(dimObj -> isOfType(dimObj, PROGRAM_ATTRIBUTE))
+            .map(TrackedEntityAttribute.class::cast)
+            .filter(TrackedEntityAttribute::isConfidentialBool)
+            .collect(Collectors.toSet());
+
+    if (!confidentialAttributes.isEmpty()) {
+      throw new IllegalQueryException(
+          new ErrorMessage(
+              ErrorCode.E7239,
+              confidentialAttributes.stream()
+                  .map(TrackedEntityAttribute::getDisplayName)
+                  .collect(Collectors.joining(", "))));
+    }
+
+    Set<DataElement> skipAnalyticsDataElements =
+        Stream.concat(
+                eventQueryParams.getItems().stream(), eventQueryParams.getItemFilters().stream())
+            .filter(item -> isOfType(item.getItem(), DATA_ELEMENT))
+            .filter(this::isSkipAnalytics)
+            .map(item -> (DataElement) item.getItem())
+            .collect(Collectors.toSet());
+
+    if (!skipAnalyticsDataElements.isEmpty()) {
+      throw new IllegalQueryException(
+          new ErrorMessage(
+              ErrorCode.E7240,
+              skipAnalyticsDataElements.stream()
+                  .map(DataElement::getDisplayName)
+                  .collect(Collectors.joining(", "))));
+    }
+  }
+
+  private static boolean isOfType(
+      DimensionalItemObject dimensionalItemObject, DimensionItemType type) {
+    return type.equals(dimensionalItemObject.getDimensionItemType());
```

Comment:
Minor point, best practice for comparing enums is to use `==` as it will give a compile type warning when comparing different types (and enum values are singletons).

Reply:
This util method could potentially also be moved to `DimensionalItemObject` as `isDimensionItemType(..)`.